src/jsdelivr.mjs

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* eslint-env serviceworker */
+
+// Headers removed by cleanupHeaders()
+const removedHeaders = [
+  'cf-cache-status',
+  'cf-ray',
+  'server',
+  'x-jsd-version',
+  'x-jsd-version-type',
+  'x-served-by',
+  'x-cache',
+];
+
+/**
+ * Removes the headers listed in removeHeaders from the Response.
+ * It does this by creating a new Response which is a copy of the
+ * original with the headers removed.
+ *
+ * @param {Response} resp the response to clean
+ * @returns the recreated, cleaned response
+ */
+function cleanupHeaders(resp) {
+  // Can't modify the response headers, so recreate a new one with the headers removed
+  const newHeaders = new Headers();
+
+  for (const kv of resp.headers.entries()) {
+    if (!removedHeaders.includes(kv[0])) {
+      newHeaders.append(kv[0], kv[1]);
+    }
+  }
+
+  const result = new Response(resp.body, {
+    headers: newHeaders,
+    status: resp.status,
+    statusText: resp.statusText,
+  });
+  return result;
+}
+
+export async function respondJsdelivr(req) {
+  const url = new URL(req.url);
+  const paths = url.pathname.split('/');
+  const beurl = new URL(paths.slice(2).join('/'), 'https://cdn.jsdelivr.net/npm/');
+  const bereq = new Request(beurl.href);
+  console.log('fetching', bereq.url);
+  const beresp = await fetch(bereq, {
+    backend: 'jsdelivr',
+  });
+  console.log('fetched', bereq.url, beresp.status, beresp.headers.get('ETag'), beresp.headers.get('Content-Length'));
+
```

Comment:
What if jsdelivr responds with a `30x`? In that case we should handle this here, so that we don't bounce over to the other CDN needlessly.

Reply:
This is now handled in the latest commit. I have also updated the existing unpkg code to handle 301, 307 and 308 redirects in addition to the 302 is already handled.